cdap-app-fabric/src/main/java/io/cdap/cdap/security/executor/SecurityContextRunnables.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.security.executor;
+
+import io.cdap.cdap.proto.security.SecurityContext;
+import io.cdap.cdap.security.spi.authentication.SecurityRequestContext;
+
+import java.util.concurrent.Callable;
+
+/**
+ * Class containing utility methods for executor thread pools.
+ */
+public class SecurityContextRunnables {
+  static Runnable wrapRunnableWithContext(Runnable runnable) {
+    SecurityContext securityContext = SecurityRequestContext.get();
+    return () -> {
+      SecurityRequestContext.set(securityContext);
+      try {
+        runnable.run();
+      } finally {
+        SecurityRequestContext.reset();
```

Comment:
instead of clearing the context, seems better to set it back to what it was before?

this is conceptually similar to what we do with the context classloader, where we have a Classloaders.setContextClassloader() that returns the original value, which we set back in a finally block.